### PR TITLE
use ===

### PR DESCRIPTION
**Pull request by cbbfcd**
thx the repo! 😄

### CODE REVIEW

File: src/core/compile.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ export const compile = (str, defs, data) => {
     return str.reduce((out, next, i) => {
         let tail = defs[i];
     
-        if (typeof defs[i] == "function") {
+        if (typeof defs[i] === "function") {
```
Comment (cristianbote) [inline src/core/compile.js]: Hey @cbbfcd, thanks for taking the time to open this! Really appreciate it!

The use of double equal, it's meant to keep the size low while maintaining proper functionality. Have you encountered issues with the double and not the triple, type checking, as well?

**Comment by Hotell** [comment]
time to add eslint for these kind of situations. so contributors will know it's ok :)

File: src/core/compile.js (hunk 1 above)
Comment (cbbfcd) [reply src/core/compile.js]: @cristianbote thx for your reply!

I fully understand the intention of using double equality, which is also safe in this scenario,
So [here](https://github.com/cristianbote/goober/blob/master/src/core/parse.js#L23) is a similar.

and it's can be replaced with `typeof val == 'object'`.

**Comment by cbbfcd** [comment]
so we can add a filesize limit check, so every commit must be checked!

File: src/core/compile.js (hunk 1 above)
Comment (cbbfcd) [reply src/core/compile.js]: in fact, I prefer to use ===, the file size can be stolen from elsewhere

Comment (cristianbote) [reply src/core/compile.js]: Hey @cbbfcd so, I've managed to shave `14B` and that meant to get rid of this exact check. There's the other one you've mentioned. If you'd like you can change that one instead.

Comment (cbbfcd) [reply src/core/compile.js]: i will make a new PR about it. thx~

**Pull request closed without merging**